juniper/src/validation/input_value.rs

```diff
@@ -70,13 +66,10 @@ fn validate_var_defs<S>(
                     errors.append(&mut unify_value(name.item, &name.start, v, &ct, schema, Path::Root));
                 }
             }
-            _ => errors.push(RuleError::new(
-                &format!(
-                    r#"Variable "${}" expected value of type "{}" which cannot be used as an input type."#,
-                    name.item, def.var_type.item,
-                ),
-                &[ name.start ],
-            )),
+            _ => panic!(
```

Comment:
I believe making this a panic undoes the fix for https://github.com/graphql-rust/juniper/issues/427 ?

Reply:
Ok, this was intended as an `unreachable!` instead of `panic!` (I will change it), since now the code is run after the validation of the document so the case should never occur. I will make sure that this is the case, and either find a unit test that checks it, or write a new one.
Thanks

Reply:
These test cases make sure that variable types are valid.

- `unknown_type_names_are_invalid` at `juniper/src/validation/rules/known_type_names.rs:90`
- `output_types_are_invalid` at `juniper/src/validation/rules/variables_are_input_types.rs:67`

It could make sense to check that `validate_var_defs` and the rest of execution is never invoked when document validation fails (in a very general sense), but beside this I don't see any meaningful way to test that the `unreachable!` is never reached...